desc/objectives/getters.py

```diff
@@ -306,11 +356,45 @@ def _get_NAE_constraints(
     return constraints
 
 
-def maybe_add_self_consistency(thing, constraints):
+def maybe_add_self_consistency(thing, constraints):  # noqa: C901
     """Add self consistency constraints if needed."""
     params = set(unique_list(flatten_list(thing.optimizable_params))[0])
 
-    # Equilibrium
+    # check which boundary mode we are in
+    if (
+        is_any_instance(constraints, FixBoundaryR)
+        and is_any_instance(constraints, FixSectionR)
+        and is_any_instance(constraints, BoundaryRSelfConsistency)
+        and is_any_instance(constraints, SectionRSelfConsistency)
+    ):
+        raise ValueError(
+            "Cannot have both FixBoundaryR and FixSectionR constraints! "
+            "The boundary condition can be LCFS or Poincare, but not both."
+        )
+
+    if is_any_instance(constraints, FixBoundaryR) and is_any_instance(
+        constraints, BoundaryRSelfConsistency
+    ):
+        mode = "lcfs"
+    elif is_any_instance(constraints, FixSectionR) and is_any_instance(
+        constraints, SectionRSelfConsistency
+    ):
+        mode = "poincare"
+    elif is_any_instance(constraints, FixBoundaryR) and not is_any_instance(
```

Comment:
isn't this sort of unnecessary since you will eror above if FixbdryR and FixSectionR are in cnostraints?